lib/Models/CsvCatalogItem.ts

```diff
@@ -141,7 +143,19 @@ export default class CsvCatalogItem
       | undefined = this.strata.get(
       automaticTableStylesStratumName
     ) as TableAutomaticStylesStratum;
-    return automaticTableStylesStratum?.discreteTimes;
+    return automaticTableStylesStratum?.discreteTimes?.reduce?.(
+      // is it correct for discrete times to remove duplicates?
+      // duplicates will mess up the indexing problem as our `<DateTimePicker />`
+      // will eliminate duplicates on the UI front, so given the datepicker
+      // expects uniques, return uniques here
```

Comment:
Perhaps we need a second getter on the `DiscretelyTimeVaryingMixin` so that we have the choice to return a deduplicated array or the original array - that might help make it slightly more explicit as to what the expected behaviour is for each of the respective getters is?

Reply:
see https://github.com/TerriaJS/terriajs/pull/4577#issuecomment-669086838